### DISCUSSION TITLE
Some general questions

### DISCUSSION THREAD

**Discussion by ResistanceIsUseless** [cea-sec/ivre#1549] [General]
I'm still trying to understand some of core philosophy behind ivre so I apologize if these are stupid questions. 

1. How come ivre doesn't automatically add results to the database? Having to manually add it makes automation slightly more complex.
2. Are there any plans to utilize the UI to create scans? It would be nice to be able to add an agent scan from the UI as distributed and its evenly split between all the available.
3. I'm still confused on how agents work. In theory could the following work:
- Create container with ivre, map, masscan ,etc
- On run container adds itself to the database as agent using `runscansagentdb`
- Agents then can take any scans that show up and place the results in a mounted fileshare
4. From a different perspective has anyone played around with ivre and flask together?
- flask creates an endpoint which runs ivre in a container
- create a load balanced / autoscaling cluster using this container
- agents output results to file share and some type of automation imports them into database

Anyways just some thoughts. I'm currently running ivre in Kubernetes after converting the docker compose to Kubernetes yaml files. It works fine with some additional configuration, but having to run scans from terminal is rough and needs scaling.

**Comment by p-l-** [comment]
Hi!
Thanks for your questions. Here are my answers:
1. That's explained in [the documentation](https://doc.ivre.rocks/en/latest/overview/principles.html). You can use the `--update-view` of the `ivre scan2db` command if you want the view to be updated with the inserted results.
2. No. I'd happily review such contributions but have no intention to code it myself.
3. You have a `ivre/agent` image where you can add targets and retrieve results.
4. I did not. I run IVRE in a Kubernetes cluster but have no need for automated scans.
Hope this helps / clarifies the current situation a bit.
